internal/controller/apiextensions/composite/transforms.go

```diff
@@ -258,11 +263,29 @@ func stringConvertTransform(input any, t *v1.StringConversionType) (any, error)
 	case v1.StringConversionTypeFromBase64:
 		s, err := base64.StdEncoding.DecodeString(str)
 		return string(s), errors.Wrap(err, errDecodeString)
+	case v1.StringConversionTypeToSHA1:
+		hash, err := stringGenerateHash(input, sha1.Sum)
+		return hex.EncodeToString(hash[:]), errors.Wrap(err, errHash)
+	case v1.StringConversionTypeToSHA256:
+		hash, err := stringGenerateHash(input, sha256.Sum256)
+		return hex.EncodeToString(hash[:]), errors.Wrap(err, errHash)
+	case v1.StringConversionTypeToSHA512:
+		hash, err := stringGenerateHash(input, sha512.Sum512)
+		return hex.EncodeToString(hash[:]), errors.Wrap(err, errHash)
 	default:
 		return nil, errors.Errorf(errStringConvertTypeFailed, *t)
 	}
 }
 
+func stringGenerateHash[THash any](input any, hashFunc func([]byte) THash) (THash, error) {
+	inputJSON, err := json.Marshal(input)
```

Comment:
Is this the first usage of generics in Crossplane 🤔

Reply:
Its a feature of the newer Go version. Is that a problem? Its only a very small function.